### PR TITLE
#159164223 Implement Google Calendar API authorization

### PR DESCRIPTION
**Pull request by OsayamenOsaretin**
#### What Does This PR Do?
This PR adds support for OAuth2 authorization for Google Calendar API.

#### Description Of Task To Be Completed
- Return authorization URL for unauthorized Users.
- Process and save credentials using the OAuth Callback URL

#### Any Background Context You Want To Provide?
This task implements OAuth2 authorization for Google Calendar API so as to allow for Invitations to be sent to Event Category subscribers when new events are created.

#### How can this be manually tested?
- Modify the .env file to match .env.sample. Add:
              
                   OAUTH2_CLIENT_ID
                   OAUTH2_CLIENT_SECRET
                   OAUTH2_PROJECT_ID
                   OAUTH2_REDIRECT_URI

- Run `tox`
- Manually try creating an event with a Calendar Authorized user, and vice versa.

#### What are the relevant github issues?
[#159164223](https://www.pivotaltracker.com/story/show/159164223)
[#159004407](https://www.pivotaltracker.com/story/show/159004407)
[#159164298](https://www.pivotaltracker.com/story/show/159164298)

### CODE REVIEW

**Comment by OsayamenOsaretin** [comment]
The tests on CircleCI are failing because we have to update the environment variables.

File: server/api/urls.py [hunk 1]
```diff
@@ -45,4 +45,7 @@
 
     url(r'^event/(?P<pk>[0-9]+)/?$', views.EventDetail.as_view(),
         name='event_detail'),
+
+    url(r'^oauthcallback/?$', views.OauthCallback.as_view(),
```
Comment (tosmak16) [inline server/api/urls.py]: @OsayamenOsaretin Please In what context or where will this be used?

Comment (OsayamenOsaretin) [reply server/api/urls.py]: This is a callback for the Oauth2 authorization flow that is called by the authorization server from Google when the user grants or doesn't grant access to our app.

Comment (tosmak16) [reply server/api/urls.py]: @OsayamenOsaretin Okay.

**Pull request merged**